automation/flexera/outdated_applied_policies/outdated_applied_policies.pt

```diff
@@ -0,0 +1,519 @@
+  end
+  result do
+    encoding "json"
+    collect jmes_path(response, "items[*]") do
+      field "id", jmes_path(col_item, "id")
+      field "href", jmes_path(col_item, "href")
+      field "name", jmes_path(col_item, "name")
+      field "description", jmes_path(col_item, "description")
+      field "category", jmes_path(col_item, "category")
+      field "created_at", jmes_path(col_item, "created_at")
+      field "created_by", jmes_path(col_item, "created_by.email")
+      field "frequency", jmes_path(col_item, "frequency")
+      field "category", jmes_path(col_item, "category")
+      field "credentials", jmes_path(col_item, "credentials")
+      field "options", jmes_path(col_item, "options")
+      field "severity", jmes_path(col_item, "severity")
+      field "skip_approvals", jmes_path(col_item, "skip_approvals")
+      field "dry_run", jmes_path(col_item, "dry_run")
+      field "running_project_ids", jmes_path(col_item, "running_project_ids")
+      field "published_template", jmes_path(col_item, "published_template")
+    end
+  end
+end
+
+datasource "ds_catalog_policies" do
+  request do
+    verb "GET"
+    host "raw.githubusercontent.com"
+    path "/flexera-public/policy_templates/master/data/active_policy_list/active_policy_list.json"
+    header "User-Agent", "RS Policies"
+  end
+  result do
+    encoding "json"
+    collect jmes_path(response, "policies[*]") do
+      field "name", jmes_path(col_item, "name")
+      field "file_name", jmes_path(col_item, "file_name")
+      field "version", jmes_path(col_item, "version")
+      field "change_log", jmes_path(col_item, "change_log")
+      field "description", jmes_path(col_item, "description")
+      field "category", jmes_path(col_item, "category")
+      field "severity", jmes_path(col_item, "severity")
+      field "readme", jmes_path(col_item, "readme")
+      field "provider", jmes_path(col_item, "provider")
+      field "service", jmes_path(col_item, "service")
+      field "policy_set", jmes_path(col_item, "policy_set")
+    end
+  end
+end
+
+datasource "ds_outdated_policies" do
+  run_script $js_outdated_policies, $ds_applied_policies, $ds_catalog_policies, $ds_policy_aggregates, $ds_self_policy, $param_ignore_list, rs_org_id, rs_project_id, policy_id, rs_optima_host
+end
+
+script "js_outdated_policies", type: "javascript" do
+  parameters "ds_applied_policies", "ds_catalog_policies", "ds_policy_aggregates", "ds_self_policy", "param_ignore_list", "rs_org_id", "rs_project_id", "policy_id", "rs_optima_host"
+  result "result"
+  code <<-'EOS'
+  tld_table = {
+    "api.optima.flexeraeng.com": "app.flexera.com",
+    "api.optima-eu.flexeraeng.com": "app.flexera.eu",
+    "api.optima-apac.flexeraeng.com": "app.flexera.au"
+  }
+
+  tld = tld_table[rs_optima_host]
+
+  aggregate_object = {}
+
+  _.each(ds_policy_aggregates, function(agg) {
+    if (agg['running_project_ids'].length == 1 && agg['running_project_ids'][0] == rs_project_id) {
```

Comment:
```suggestion
    if (agg['running_project_ids'] && agg['running_project_ids'].length == 1 && agg['running_project_ids'][0] == rs_project_id) {
```

Ran this Vestas's tenant and got the below error, so added a condition that checks if the `running_project_id` field exists in the object:
![image](https://github.com/flexera-public/policy_templates/assets/92175447/41f9781e-1eb1-4674-b20b-354d9cd1dd80)